frontend/src/pages/NicknamePage/index.tsx

```diff
@@ -31,26 +29,29 @@ const NicknamePage = ({ gameCode }: Props) => {
       const gameCodeIsValid = res.success;
       if (gameCodeIsValid) {
         if (playerId && token) {
-          // TODO try connect to socket.io
-          // No way to check whether playerId is for the current gameCode
-        } else {
-          // Stay on this page
+          socket.auth = { gameCode, playerId, token };
+          socket.connect();
+          // If the connection here, it means the playerId is invalid so need to remove it
+          // It reruns this useEffect since playerId changed, but it does not connect again because playerId is falsy
+          socket.on("connect_error", () => {
+            setPlayerId("");
+            setToken("");
+          });
```

Comment:
I wonder if this should be moved to GameRouter to be included in the useEffect() & be unmounted 🤔